diff --git a/includes/media.admin.inc b/includes/media.admin.inc
index 88b88b9..10756ba 100644
--- a/includes/media.admin.inc
+++ b/includes/media.admin.inc
@@ -62,7 +62,23 @@ function media_import_validate($form, &$form_state) {
       form_set_error('directory', t('The provided directory does not exist.'));
     }
     $pattern = !empty($pattern) ? $pattern : '*';
-    $files = glob("$directory/$pattern");
+
+    $dirs = array();
+    $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($directory), RecursiveIteratorIterator::SELF_FIRST);
+    foreach ($iterator as $file) {
+      if ($file->isDir()) {
+        $path = $file->getRealpath();
+        if (!in_array($path, $dirs) && (strlen(realpath($directory)) <= strlen($path))) {
+          array_push($dirs, $path);
+        }
+      }
+    }
+    $files = array();
+    foreach ($dirs as $dir) {
+      $files = array_merge($files, glob("$dir/$pattern"));
+    }
+    $files = array_diff($files, $dirs);
+
     if (empty($files)) {
       form_set_error('pattern', t('No files were found in %directory matching %pattern', array('%directory' => $directory, '%pattern' => $pattern)));
     }
